File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.cpp
class Solution {
public:
    string reverseParentheses(string s) {
        int n=s.size();
        string ans;
        stack<int>st;
        for(int i=0;i<n;i++)
        {
            if(s[i]=='(')st.push(ans.size());
            else if(s[i]==')')
            {
                reverse(ans.begin()+st.top(),ans.end());
                st.pop();
            }
            else
                ans+=s[i];
        }
        return  ans;
    }
};

File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.java
class Solution {
    public String reverseParentheses(String s) {
        Stack<String>st=new Stack<>();
        for(int i=0;i<s.length();i++)
        {
            if(s.charAt(i)==')')
            {
                String newStr="";
                while(!st.peek().equals("("))
                {
                    newStr+=new StringBuilder(st.pop()).reverse().toString();
                }
                st.pop();
                st.push(newStr);
            }
            else
            {
                st.push(s.substring(i,i+1));
            }
        }
        String newStr="";
        while(!st.empty())
        {
            newStr+=new StringBuilder(st.pop()).reverse().toString();;
        }
        return new StringBuilder(newStr).reverse().toString();
    }
}